service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java

```diff
@@ -1,209 +1,288 @@
-    this.transactions = createTransactions();
+    HashMap<String, PsuData> map = new HashMap<>();
+
+    PsuData psuSuccessful = initPsuSuccessfull();
+
+    map.put(psuSuccessful.getPsuId(), psuSuccessful);
+
+    this.psuMap = map;
   }
 
-  public List<Transaction> getTransactions(String accountId) {
-    return new ArrayList<>(transactions.get(accountId).values());
+  public PsuData getPsu(String psuId) {
+    return psuMap.get(psuId);
   }
 
-  public Transaction getTransactionDetails(String accountId, String transactionId) {
-    return transactions.get(accountId).get(transactionId);
+  /**
+   * Returns Psu-Id by Iban.
+   *
+   * @param iban Iban
+   * @return String Psu-Id
+   */
+  public String getPsuByIban(String iban) {
+    for (Map.Entry<String, PsuData> currentPsu : psuMap.entrySet()) {
+      String currentPsuId = currentPsu.getValue().getPsuId();
+
+      List<Account> result = currentPsu.getValue().getAccounts().values().stream()
+          .filter(account -> account.getIban().equals(iban))
+          .collect(Collectors.toList());
+
+      if (result.size() > 0) {
+        return currentPsuId;
+      }
+    }
+    return "";
   }
 
-  public List<Account> getAccounts() {
-    return new ArrayList<>(accounts.values());
+  /**
+   * Returns AccountId by Iban.
+   *
+   * @param psuId psuId
+   * @param iban  Iban
+   * @return AccountId
+   */
+  public String getAccountIdByIban(String psuId, String iban) {
+    HashMap<String, Account> map = psuMap.get(psuId).getAccounts();
+
+    Account result = map.values().stream()
+        .filter(account -> account.getIban().equals(iban))
+        .collect(Collectors.toList()).get(0);
+
+    return result.getAccountId();
   }
 
-  public Account getAccountDetails(String id) {
-    return accounts.get(id);
+  public List<Account> getAccounts(String psuId) {
+    return new ArrayList<>(psuMap.get(psuId).getAccounts().values());
   }
 
-  private HashMap<String, Account> createAccounts() {
-    HashMap<String, Account> accounts = new HashMap<>();
-    Account giroAccount = new Account(
-        ACCOUNT_ID_GIRO,
-        IBAN_GIRO,
-        EUR,
-        BigDecimal.valueOf(1500)
-    );
+  public Account getDistinctAccount(String psuId, String accountId) {
+    return psuMap.get(psuId).getAccounts().get(accountId);
+  }
 
-    Account savingsAccount = new Account(
-        ACCOUNT_ID_SAVINGS,
-        IBAN_SAVINGS,
-        EUR,
-        BigDecimal.valueOf(2300)
-    );
+  /**
+   * Returns List of Accounts depending on parameter input.
+   *
+   * @param psuId Psu-Id
+   * @param ibans List of Ibans
+   * @return List of Accounts
+   */
+  public List<Account> getRequestedAccounts(String psuId, List<String> ibans) {
+    List<Account> requestedAccounts = new ArrayList<Account>() {
+    };
 
-    accounts.put(giroAccount.getAccountId(), giroAccount);
-    accounts.put(savingsAccount.getAccountId(), savingsAccount);
-    return accounts;
+    for (int i = 0; i < ibans.size(); ++i) {
```

Comment:
- get all accounts for PSU
- remove every account that's not in ibans

Reply:
Done